src/pages/workspaces/workspace/applications/AppLauncher.js

```diff
@@ -15,13 +15,27 @@ import * as Utils from 'src/libs/utils'
 import { wrapWorkspace } from 'src/pages/workspaces/workspace/WorkspaceContainer'
 
 
+const getSparkInterfaceSource = (proxyUrl, sparkInterface) => {
+  console.assert(_.endsWith('/jupyter', proxyUrl), 'Unexpected ending for proxy URL')
+  const proxyUrlWithlastSegmentDropped = _.flow(_.split('/'), _.dropRight(1), _.join('/'))(proxyUrl)
+  return `${proxyUrlWithlastSegmentDropped}/${sparkInterface}`
+}
+
+const getApplicationIFrameSource = (proxyUrl, application, sparkInterface) => {
+  return Utils.switchCase(application,
+    ['terminal', () => `${proxyUrl}/terminals/1`],
+    ['spark', () => getSparkInterfaceSource(proxyUrl, sparkInterface)],
+    [Utils.DEFAULT, () => proxyUrl] // RStudio
```

Comment:
I'd prefer to explicitly have a branch for `tools.label.RStudio`, and have default give some sort of error.